### PR TITLE
Set setuptools_scm fallback_version to follow  PEP440

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Address comment https://github.com/GenericMappingTools/pygmt/pull/1784#issuecomment-1148925436 and https://github.com/GenericMappingTools/pygmt/pull/1848#discussion_r891579228.

To test it, try:
```
pip install https://github.com/GenericMappingTools/pygmt/archive/refs/heads/fallback_version.zip
```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ build-backend = "setuptools.build_meta"
 
 [tool.setuptools_scm]
 local_scheme = "node-and-date"
-fallback_version = "unknown"
+fallback_version = "9999.9999.9999"
```
Comment (weiji14) [inline pyproject.toml:7]: A quick search using https://github.com/search?l=TOML&p=1&q=tool.setuptools_scm+fallback_version&type=Code shows a lot of incosistency across projects. But I did notice someone use something like `0.0.0+unknown`. I think it should be compatible with PEP440 following https://peps.python.org/pep-0440/#local-version-identifiers, but need to test first
```suggestion
fallback_version = "0.0.0+unknown"
```

Comment (seisman) [reply pyproject.toml:7]: My initial thought is a version string like `0.0.0`, but my concern is, if someone develops a package that requires PyGMT greater than a specific version (for example, checking `pygmt.__version__ >= "0.6.0"`), he won't be able to use the PyGMT with a version string `0.0.0` or `0.0.0+unknown`.

**Comment by weiji14** [comment]
Cool, seems to work ok now, `pip install https://github.com/GenericMappingTools/pygmt/archive/refs/heads/fallback_version.zip`:

```
Collecting https://github.com/GenericMappingTools/pygmt/archive/refs/heads/fallback_version.zip
  Downloading https://github.com/GenericMappingTools/pygmt/archive/refs/heads/fallback_version.zip
     | 3.9 MB 4.2 MB/s 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: numpy>=1.19 in /home/username/mambaforge/envs/pygmt/lib/python3.9/site-packages (from pygmt==999.999.999+unknown) (1.22.4)
Collecting netCDF4
  Using cached netCDF4-1.5.8-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.7 MB)
Requirement already satisfied: xarray in /home/username/mambaforge/envs/pygmt/lib/python3.9/site-packages (from pygmt==999.999.999+unknown) (2022.3.0)
Requirement already satisfied: packaging in /home/username/mambaforge/envs/pygmt/lib/python3.9/site-packages (from pygmt==999.999.999+unknown) (21.3)
Requirement already satisfied: pandas in /home/username/mambaforge/envs/pygmt/lib/python3.9/site-packages (from pygmt==999.999.999+unknown) (1.4.2)
Collecting cftime
  Using cached cftime-1.6.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (239 kB)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /home/username/mambaforge/envs/pygmt/lib/python3.9/site-packages (from packaging->pygmt==999.999.999+unknown) (3.0.9)
Requirement already satisfied: pytz>=2020.1 in /home/username/mambaforge/envs/pygmt/lib/python3.9/site-packages (from pandas->pygmt==999.999.999+unknown) (2022.1)
Requirement already satisfied: python-dateutil>=2.8.1 in /home/username/mambaforge/envs/pygmt/lib/python3.9/site-packages (from pandas->pygmt==999.999.999+unknown) (2.8.2)
Requirement already satisfied: six>=1.5 in /home/username/mambaforge/envs/pygmt/lib/python3.9/site-packages (from python-dateutil>=2.8.1->pandas->pygmt==999.999.999+unknown) (1.16.0)
Building wheels for collected packages: pygmt
  Building wheel for pygmt (pyproject.toml) ... done
  Created wheel for pygmt: filename=pygmt-999.999.999+unknown-py3-none-any.whl size=346893 sha256=0bcf92ecfbd89c9cb22e42497a222472edd5a521cc56eaca75115482b20c67b7
  Stored in directory: /tmp/pip-ephem-wheel-cache-0kntmfs0/wheels/bc/33/c2/8f3051bbe2f3d0737fe8d1dc5baa0e42e04ba7cd1c44947942
Successfully built pygmt
Installing collected packages: cftime, netCDF4, pygmt
Successfully installed cftime-1.6.0 netCDF4-1.5.8 pygmt-999.999.999+unknown
```

**Review by weiji14 — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (seisman) [reply pyproject.toml]: It works well. Marking the PR as ready for review.

**Pull request merged**